sbi/neural_nets/classifier.py

```diff
@@ -113,6 +115,11 @@ def build_linear_classifier(
     Returns:
         Neural network.
     """
+    # Initialize default embedding net
+    if 'DefaultEmbeddingNet' in str(embedding_net_x):
```

Comment:
hm yeah this is not great. How about `if isinstance(embedding_net_x, DefaultEmbeddingNet)`?